### PR TITLE
164826821 random fixes

### PR DESCRIPTION
**Pull request by rckopteri**
# Fixed
* Sea route on-blur when save is clicked without clicking "off" from homepage input field
* Admin sea-route-list last active day when only added calendar date is given and no calendar rule

### CODE REVIEW

File: ote/src/cljs/ote/views/admin/sea_routes.cljs [hunk 1]
```diff
@@ -64,7 +64,9 @@
                                (if (true? ((weekday-index-to-keyword last-weekday-index) route))
                                  (* -1 index)
                                  (recur (inc index) (dec last-weekday-index)))
-                               nil))]
+                               ; return 0 days only if to-date and weekday are given.
+                               (when (and to-date weekday)
+                                 0)))]
     (if (not (nil? last-weekday-count))
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/admin/sea_routes.cljs]:
```suggestion
(when last-weekday-count
      (time/native->date
        (t/plus (time/native->date-time to-date) (t/days last-weekday-count))))
```

File: ote/src/cljs/ote/views/admin/sea_routes.cljs [hunk 2]
```diff
@@ -64,7 +64,9 @@
                                (if (true? ((weekday-index-to-keyword last-weekday-index) route))
                                  (* -1 index)
                                  (recur (inc index) (dec last-weekday-index)))
-                               nil))]
+                               ; return 0 days only if to-date and weekday are given.
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/admin/sea_routes.cljs]:
```suggestion
                               ;; return 0 days only if to-date and weekday are given.
```
https://guide.clojure.style/#two-semicolons-for-code-fragment

+Add a "why" into the comment if reasonable.

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 3]
```diff
@@ -339,6 +339,8 @@
   UpdateOperatorHomepageResponse
   (process-event [{response :response} app]
     (-> app
+        ; clear operator homepage from route - to prevent unnecessary save
+        (dissoc [:route :operator-homepage])
```
Comment (philosotractor) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: Or rather select-keys (specql/columns <tablename> c) in service side.

File: ote/src/cljs/ote/views/admin/sea_routes.cljs (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/admin/sea_routes.cljs]: Replaced with when-not

File: ote/src/cljs/ote/views/admin/sea_routes.cljs (hunk 2 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/admin/sea_routes.cljs]: why added

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs (hunk 3 above)
[the remark this answers is not in the document]
Comment (rckopteri) [reply ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: Maybe in distant future. Actually :operator-homepage key is removed in every case, so it won't cause problems but it is not a robust way to handle things.
We should refactor all our form-data codes in same time.

[the remark this answers is not in the document]
Comment (philosotractor) [reply ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: While this of course does not block this PR,
I might disagree on the reasoning. Adding such a select-keys to service side is a simple and safe one-liner and can be implemented incrementally and does not require any other refactoring :)

**Pull request merged**